policy/policy.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package policy
+
+import (
+	"errors"
+	"fmt"
+	"github.com/romana/core/common"
+	"github.com/romana/core/tenant"
+	"log"
+	//	"net"
+)
+
+// Policy provides Policy service.
+type PolicySvc struct {
+	config common.ServiceConfig
+}
+
+const (
+	infoListPath = "/info"
+)
+
+func (policy *PolicySvc) Routes() common.Routes {
+	routes := common.Routes{
+		common.Route{
+			Method:          "POST",
+			Pattern:         "/policies",
+			Handler:         policy.addPolicy,
+			MakeMessage:     func() interface{} { return &common.Policy{} },
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:          "DELETE",
+			Pattern:         "/policies/{id}",
+			Handler:         policy.deletePolicy,
+			MakeMessage:     nil,
+			UseRequestToken: false,
+		},
+	}
+	return routes
+}
+
+// handleHost handles request for a specific host's info
+func (policy *PolicySvc) addPolicy(input interface{}, ctx common.RestContext) (interface{}, error) {
+	policyDoc := input.(*common.Policy)
```

Comment:
This is a very long function, which could benefit greatly from a few comments throughout its body.